src/editors/containers/TextEditor/TextEditor.jsx

```diff
@@ -42,11 +45,18 @@ export const TextEditor = ({
 }) => {
   const { isOpen, openModal, closeModal } = modalToggle();
 
+  // selected image file reference data object.
+  // existance of this field determines which child modal is displayed
+  const [imageSelection, setImageSelection] = React.useState(null);
+
```

Comment:
I think the clearSelection definition should be put here as well so we have the logic group together.

Reply:
Not a bad shout, but I would rather be in favor of removing `clearSelection` than adding another prop to pass around, and just calling `setImageSelection(null)` where we used to use `clearSelection` as they are the same thing.

Reply:
That sounds fine to me as well! My concern was only grouping the selection logic together.